Labeler workflow fails on fork PRs: Resource not accessible by integration

## Summary

The `Labeler` workflow (`.github/workflows/labeler.yml`) fails on PRs from forks with `Resource not accessible by integration`, even though the workflow declares `pull-requests: write`.

## Repro

- PR #75 (from fork `SAY-5/homestead`)
- Run: https://github.com/haydenk/homestead/actions/runs/24820507171

```
The action requires write permission to add labels to pull requests.
##[error]Resource not accessible by integration
```

## Root cause

On `pull_request` events triggered from forks, `GITHUB_TOKEN` is forced read-only regardless of the workflow's `permissions:` block. This is a GitHub security policy — the workflow file's permissions block can't override it.

## Fix

Switch the PR trigger from `pull_request` to `pull_request_target`. The labeler action only inspects changed file paths via the API — it does not check out or execute fork code, so this is safe.

```diff
 on:
-  pull_request:
+  pull_request_target:
     types: [opened, synchronize, reopened]
   issues:
     types: [opened, edited]
```

`pull_request_target` always loads the workflow file from the base branch, so once merged this applies to all future fork PRs (and to PR #75 once its branch is pushed to or it's reopened).

## Notes

- The `label-issue` job (uses `issues` event) is unaffected.
- No changes needed in any fork — this is a base-repo-only fix.

--- a/.github/workflows/labeler.yml
+++ b/.github/workflows/labeler.yml
@@ -1,7 +1,7 @@
 name: Labeler
 
 on:
-  pull_request:
+  pull_request_target:
     types: [opened, synchronize, reopened]
   issues:
     types: [opened, edited]

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- **Labeler workflow on fork PRs** — Switched the `Labeler` workflow's PR trigger from `pull_request` to `pull_request_target` so `GITHUB_TOKEN` retains write permissions on PRs from forks; the action only reads changed file paths and does not check out fork code (#77)
+
 
 ## [v0.3.0] - 2026-03-07
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+
+## [v0.3.1] - 2026-04-26
+
 ### Fixed
 
 - **Labeler workflow on fork PRs** — Switched the `Labeler` workflow's PR trigger from `pull_request` to `pull_request_target` so `GITHUB_TOKEN` retains write permissions on PRs from forks; the action only reads changed file paths and does not check out fork code (#77)

